recipes/linecache2/test_linecache.patch

```diff
@@ -0,0 +1,109 @@
+--- linecache2/tests/test_linecache.py
++++ linecache2/tests/test_linecache.py
+@@ -13,8 +13,6 @@
+ NONEXISTENT_FILENAME = FILENAME + '.missing'
+ INVALID_NAME = '!@$)(!@#_1'
+ EMPTY = ''
+-TESTS = 'inspect_fodder inspect_fodder2 mapping_tests'
+-TESTS = TESTS.split()
+ TEST_PATH = os.path.dirname(__file__)
+ MODULES = "linecache abc".split()
+ MODULE_PATH = os.path.dirname(FILENAME)
+@@ -37,7 +35,67 @@
+
+ SOURCE_3 = '''
+ def f():
+-    return 3''' # No ending newline
++    return 3'''  # No ending newline
++
++
++class TempFile:
++
++    def setUp(self):
++        super().setUp()
++        with tempfile.NamedTemporaryFile(delete=False) as fp:
++            self.file_name = fp.name
++            fp.write(self.file_byte_string)
++        self.addCleanup(support.unlink, self.file_name)
++
++
++class GetLineTestsGoodData(TempFile):
++    # file_list   = ['list\n', 'of\n', 'good\n', 'strings\n']
++
++    def setUp(self):
++        self.file_byte_string = ''.join(self.file_list).encode('utf-8')
++        super().setUp()
++
++    def test_getline(self):
++        with open(self.file_name) as fp:
++            for index, line in enumerate(fp):
++                if not line.endswith('\n'):
++                    line += '\n'
++
++                cached_line = linecache.getline(self.file_name, index + 1)
++                self.assertEqual(line, cached_line)
++
++    def test_getlines(self):
++        lines = linecache.getlines(self.file_name)
++        self.assertEqual(lines, self.file_list)
++
++
++class GetLineTestsBadData(TempFile):
++    # file_byte_string = b'Bad data goes here'
++
++    def test_getline(self):
++        self.assertRaises((SyntaxError, UnicodeDecodeError),
++                          linecache.getline, self.file_name, 1)
++
++    def test_getlines(self):
++        self.assertRaises((SyntaxError, UnicodeDecodeError),
++                          linecache.getlines, self.file_name)
++
++
++class EmptyFile(GetLineTestsGoodData, unittest.TestCase):
++    file_list = []
++
++
++class SingleEmptyLine(GetLineTestsGoodData, unittest.TestCase):
++    file_list = ['\n']
++
++
++class GoodUnicode(GetLineTestsGoodData, unittest.TestCase):
++    file_list = ['á\n', 'b\n', 'abcdef\n', 'ááááá\n']
++
++
++class BadUnicode(GetLineTestsBadData, unittest.TestCase):
++    file_byte_string = b'\x80abc'
```

Comment:
Ah, whoops. Yeah - that's taken from the patch mentioned in [this comment](https://github.com/testing-cabal/linecache2/issues/2#issuecomment-176871454) That patch was applied to a different commit than the current one in the base repo, so I ported those changes into a new file. That could hasn't been thoroughly reviewed; rather, I was trusting that the various adaptations were indeed necessary for providing tests missing from the original. The fact that the original patches weren't sufficient surprised me, and resulted in the commenting spree in `test_linecache.py`

So: before passing this on to upstream we should establish that everything in the`inspect_fodder.py` patch is still essential to the build.